### DISCUSSION TITLE
Any reason why `pull` for ReplicatedCollection overwrites the whole collection?

### DISCUSSION THREAD

**Discussion by Blankeos** [maxnowack/signaldb#791] [Q&A]
If there's no changes, can't it just, not do anything?

Everytime I refresh the page, it kind of always assumes that it should revalidate everything stored. I think that's a bit problematic. But not sure if that's the right approach.
  - Let's say I made some changes offline. Then went online, it will revalidate my collection and my pushed changes won't be there. Not sure.

Not sure if I'm doing it right.

I tried this by doing:

```
pull: {
   
   const changes = await someApI();
   
   return {
      changes: {
         added: [],
         modified: [].
         deleted: [],
      }
   }
}
```

https://github.com/maxnowack/signaldb/assets/38070918/1aad5d6b-e2c1-4eb0-9fe0-82ac9a913e27

**Comment by maxnowack** [comment]
The `ReplicatedCollection` doesn't have any automatic error handling. So if your `push` is failing because you're offline, you have to handle it by yourself (e.g. caching items and transferring them when online again). I know that this is currently not very convenient and it will be changed in a later version. The current implementation is the very first version of the `ReplicationCollection` and has only very basic functionality.

**Reply by Blankeos** [reply]
Thanks for the quick responses Max! Really appreciate it! That's good to know at least.

**Reply by maxnowack** [reply]
You're welcome! Glad to help! 🙂
